NGS2_WITNESS_Cycle1_experiment1_prep.R

```diff
@@ -0,0 +1,422 @@
+
+# calculate actor moves by round
+round_changes <- changes_by_round(chng, conn)
+nbr_connections_round <- calculate_connections(round_changes)
+
+# finish off the `cooperation` dataset
+coop <- mapply(function(x, y, exp) {
+    if(nrow(y) > 0) {
+        return(merge(x, y, by = c('round', 'pid')))
+    } else {
+        conns <- dcast(subset(exp, event == 'Connected'), id ~ data.name,
+                       value.var = 'data.value')
+        x$num_neighbors <- NA
+        for(i in 1:max(x$round)) {
+            x <- get_neighbor_number(x, conns, i)
+        }
+        return(x[, c('round', 'pid', 'action', 'group_size',
+                     'previous_decision', 'num_neighbors')])
+    }
+}, coop, nbr_connections_round, exp1, SIMPLIFY = FALSE)
+
+# add in parameters from dataset for completeness
+coop <- do.call(rbind, mapply(function(x, name, exp) {
+    exp_cond <- as.numeric(exp$data.value[exp$event == 'initParameters' &
+                                          exp$data.name == 'k'])
+    x$session <- name
+    x$condition <- ifelse(exp_cond == 0, 'Static',
+                          ifelse(exp_cond == .1, 'Viscous',
+                          ifelse(exp_cond == .3, 'Fluid',
+                          ifelse(exp_cond == 1, 'Random', 'Other'))))
+    return(x)
+}, coop, as.list(names(coop)), exp1, SIMPLIFY = FALSE))
+
+# add in empanelment id for non-bots
+coop <- merge_with_empanelment(coop, empanel_bb_xwalk, 'pid')
+
+# output dataset to disk
+write.csv(coop[order(coop$session, coop$round, coop$pid), COOP_VAR_ORDER],
+          file = 'NGS2-Cycle1-Experiment1/cooperation_exp1.csv',
+          row.names = FALSE, na='')
+
+# rewire dataset
+# start building `rewire` output file
+rewire <- build_rewire(chng)
+
+# add respondent actions to `rewire` dataset
+rewire <- add_actions(rewire, act, focus = 'player')
+rewire <- add_actions(rewire, act, focus = 'alter')
+
+rewire <- lapply(rewire, function(x) {
+    if(nrow(x) > 0) {
+        x$state <- apply(x[, grep('action$', names(x))], 1, function(y) {
+            paste0(y, collapse = '')
+        })
+        x$CC <- ifelse(x$state == 'CC', 1, 0)
+        x$DD <- ifelse(x$state == 'DD', 1, 0)
+        x$other_d <- ifelse(x$alteraction == 'D', 1, 0)
+        return(x)
+    } else {
+        return(data.frame())
+    }
+})
+
+# add in previous and current ties
+rewire <- add_ties(rewire, conn, round_changes)
+
+# add oin parameters from dataset for completeness
+rewire <- do.call(rbind, mapply(function(x, name) {
+    if(nrow(x) > 0) {
+        x$session <- name
+    }
+    return(x)
+}, rewire, as.list(names(rewire)), SIMPLIFY = FALSE))
+
+# add in empanelment id for non-bots
+rewire <- merge_with_empanelment(rewire, empanel_bb_xwalk, 'playerid',
+                                 rename = TRUE, rename_var = 'player_empanel_id')
+rewire <- merge_with_empanelment(rewire, empanel_bb_xwalk, 'alterid',
+                                 rename = TRUE, rename_var = 'alter_empanel_id')
+
```

Comment:
- Drop failed "e1.1_2017-08-21" session, including "other" conditions

rewire <- subset (rewire, session!="e1.1_2017-08-21")

- Drop tiny "e1.4_2017-08-31" session, with only three players

rewire <- subset (rewire, session!="e1.4_2017-08-31")

Reply:
See comment above -- I'm okay with this, as long as we make it condition-based.

Reply:
Minimum number of rounds = 2
Minimum number of players = 10

Reply:
This has been addressed with commit 8e90ef.